perfkitbenchmarker/linux_benchmarks/cloud_spanner_ycsb_benchmark.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2017 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Run YCSB benchmark against Google Cloud Spanner
+
+Before running this benchmark, you have to download your JSON
+service account credential file to local machine, and pass the path
+via 'cloud_spanner_credential_file' parameters to PKB.
+
+By default, this benchmark provision 1 single-CPU VM and spawn 1 thread
+to test Spanner.
+"""
+
+import getpass
+import logging
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.linux_packages import ycsb
+from perfkitbenchmarker.providers.gcp import gcp_spanner
+
+BENCHMARK_NAME = 'cloud_spanner_ycsb'
+BENCHMARK_CONFIG = """
+cloud_spanner_ycsb:
+  description: >
+      Run YCSB against Google Cloud Spanner.
+      Configure the number of VMs via --num-vms.
+  vm_groups:
+    default:
+      vm_spec: *default_single_core
+      vm_count: 1
+  flags:
+    gcloud_scopes: >
+      https://www.googleapis.com/auth/spanner.admin
+      https://www.googleapis.com/auth/spanner.data"""
+
+# As of May 2017, cloud spanner is not included in the latest YCSB release
+# 0.12.0. You can build your own release binary and feed it in with
+# --cloud_spanner_custom_ycsb_release=<url>. You are welcome to use a pre-build
+# binary located at
+# https://storage.googleapis.com/cloud-spanner-ycsb-custom-release/
+# ycsb-cloudspanner-binding-0.13.0.tar.gz
+YCSB_BINDING_TAR_URL = ('https://github.com/brianfrankcooper/YCSB/releases'
+                        '/download/0.13.0/'
+                        'ycsb-cloudspanner-binding-0.13.0.tar.gz')
+REQUIRED_SCOPES = (
+    'https://www.googleapis.com/auth/spanner.admin',
+    'https://www.googleapis.com/auth/spanner.data')
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string('table',
```

Comment:
This is spanner specific. Some storage, such as bigtable, are just key-value-pair storage and don't have a concept of table. Will rename this to cloud_spanner_table.